### ISSUE TITLE
Update existing System class doxygen to provide block diagram schematics

### ISSUE THREAD

**Issue by RussTedrake** [RobotLocomotion/drake#9496]
using the capability provided by #9490 and documented (once that PR lands) at http://drake.mit.edu/doxygen_instructions.html?highlight=doxygen .

- [x] Primitives #16128
- [x] Controllers #16193
- [x] Estimators
- [x] Sensors #16194
~~- [ ] Automotive~~
- [x] Manipulation #16195
- [x] Message Passing #16196
- [x] MultiBodyPlant #16200
- [x] SceneGraph #16202
- [x] Visualization #16203
- [x] Examples #16204

cc @SeanCurtis-TRI 

Loose ends turned up in review:
- [x] give beam_model.h a @code example showing AddRandomInputs() usage #16282

**Comment by RussTedrake** [comment]
Note: consider using an image as the system label in some of the primitive examples, to show what's possible/acceptable.

**Comment by rpoyner-tri** [comment]
I spent a bit of time fiddling with possible ways of embedding images. Unless I'm missing something, embedding an HTML <img> tag was the intention. That will work, but is not very readable in source code, and raises the new issue of how to host the image data, and whether it needs additional handling at doc-build time. I'm punting for now.

**Comment by rpoyner-tri** [comment]
For Diagrams that do not have explicit names set in Export{In,Out}put(), the generated names are unusable for Get{In,Out}putPort(str), since a pointer value in hex gets encoded into the name. Here are some possibilities for documenting them in schematics:

* don't bother creating a schematic (status quo for many Diagrams)
* create a schematic, and indicate by color/typography that the "port names" are actually method calls
* create a schematic, and also introduce string names and the Export*() calls (possibly a breaking change??)

**Comment by jwnimmer-tri** [comment]
> ... introduce string names and the Export*() calls (possibly a breaking change??)

I don't think that would be considered as a breaking change (at least, when the sub-systems did not have names assigned).  Can you link to a specific example or two, so we can consider the concrete examples?

**Comment by rpoyner-tri** [comment]
Immediate example is `PidControlledSystem` -- no schematic, and no names in current `master`.

**Comment by jwnimmer-tri** [comment]
In that case, if the `plant` was previously given a `set_name`, then the wrapped names would have been predictable (not hex), so the change would technically breaking.  Still probably worth making that change, though.  For that system, I think that the diagram's input and output port names should be identical to the wrapped plant's port names (with no extra prefix or suffix) -- except for the "desired state" input port, which we'll need to choose a more careful name for.

**Comment by rpoyner-tri** [comment]
Copying here for posterity some of the rules/goals agreed on during early PRs here.

 * list port names that can be used as arguments to Get{In,Out}putPort(str)
 * indicate with color ports that may not always be present
 * move other port annotations to the surrounding text
 * for classes that inherit ports:
   * make sure the base classes have diagrams
   * paste/rename diagrams into the derived classes
 * be consistent in the above guidelines

**Comment by rpoyner-tri** [comment]
Another note toward some kind of schematic style guide: color, type style.

Many diagrams use gray to indicate ports that may not exist in some circumstance.
Mbp uses green to indicate ports used with SceneGraph.
Mbp uses gray+italic to indicate variable substitution to produce a group of names.
Manipulation station uses orange+bold to indicate ports that only exist in simulation, but not on a real robot.

Some diagrams use gray+italic on ellipses -- not sure if that is even visible.

What shall we use for optional ports, which always exist, but may be left unconnected? I had a thought of using turquoise, but that choice is arbitrary. I'll have to try it to see how it looks rendered.

**Comment by sherm1** [comment]
For ports always present but that don't have to be connected, I'd be inclined just to render them black like other always-extant ports, but mention in the surrounding text that you aren't obligated to connect them. If you do use a different color I think you'll still have to say in the text what that color means -- I doubt many people will remember.

**Comment by rpoyner-tri** [comment]
Given that we abandoned the idea of colorizing optional ports, I punted the proposed review pass (which was my main motivation for that). We will likely be back through these improving port names if various #12214 fixes land anyway.

Declaring victory. Thanks all, and happy holidays!

**Comment by RussTedrake** [comment]
Woohoo!  Thank you!
